### ISSUE TITLE
Explore having a middleware that allows sharing of server processes between cable and app servers

### ISSUE THREAD

**Issue by dhh** [rails/actioncable#15]
MessageBus has a [rack middleware](https://github.com/SamSaffron/message_bus/blob/master/lib/message_bus/rails/railtie.rb#L16) that allows you to deal with WebSocket requests inside the same server process as the rest of the application. This means not having to run two separate processes, at least if you can run your application under a threaded server like Puma.

Whether you want to do that in production or not, it seems like this would be a better fit for development.

**Comment by rosenfeld** [comment]
Just so that you know, I'd be interested in running this on production as we already deploy with Puma and this would simplify things on our side and would be just good enough for our application requirements... I was going to suggest this but I'm glad you created this feature request yourself :)

**Comment by tagrudev** [comment]
I was looking in the examples and I was wondering exactly the same - why would I need another server if I am already running a threaded server like Puma?

**Comment by jmbejar** [comment]
FYI I run an experiment in order to test an example application using ActionCable on Heroku.

I needed to have it working on an unique puma server, so I created a middleware that is very simple:
https://github.com/jmbejar/actioncable-chat-example/commit/b7286a33d6d9d474a778466ff18b7717e5813ae1

The approach is based on https://devcenter.heroku.com/articles/ruby-websockets.

I have the feeling that is not a good choice for production because I would expect a drop in terms of performance, at least for websockets messages between server-client (it's just a feeling, I didn't run any benchmark to confirm that).

In the other hand, it looks like a good option to consider to run acceptance tests with capybara + selenium or similar, since they rely on the existence of an unique web server.
